pkg/chunkenc/gzip.go

```diff
@@ -477,73 +468,109 @@ func (li *listIterator) Close() error   { return nil }
 func (li *listIterator) Labels() string { return "" }
 
 type bufferedIterator struct {
-	s *bufio.Reader
+	s      *bufio.Reader
+	reader CompressionReader
+	pool   CompressionPool
 
-	curT   int64
-	curLog string
+	cur logproto.Entry
 
 	err error
 
-	buf    []byte // The buffer a single entry.
-	decBuf []byte // The buffer for decoding the lengths.
+	buf    *bytes.Buffer // The buffer a single entry.
+	decBuf []byte        // The buffer for decoding the lengths.
+
+	closed bool
+
+	filter logql.Filter
 }
 
-func newBufferedIterator(s *bufio.Reader) *bufferedIterator {
+func newBufferedIterator(pool CompressionPool, b []byte, filter logql.Filter) *bufferedIterator {
```

Comment:
CompressionPool is not global because depending on the chunk encoding it could be another one, but at the end of the day, the gzip which is globlab is used

Reply:
> Can we combine both into a single thing and pass it in removing globals

Yes, I don't want globals :)

Reply:
I don't see a path to achieve this without making the code way more complex because in this package there is nothing long lived enough to avoid global for pools.

Reply:
For me the simplicity out-weight the problem of using global. I think it is pretty normal to have a sync.Pool global.

Reply:
Are you ok to drop this one ?